lints/lint_mp_allowed_rsa_keys_modulus.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/********************************************************************
+Section 5.1 - Algorithms
+RSA keys whose modulus size in bits is divisible by 8, and is at least 2048.
+********************************************************************/
+
+package lints
+
+import (
+	"crypto/rsa"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type allowedRSAKeyModulus struct{}
+
+func (l *allowedRSAKeyModulus) Initialize() error {
+	return nil
+}
+
+func (l *allowedRSAKeyModulus) CheckApplies(c *x509.Certificate) bool {
+	if c.PublicKeyAlgorithm == x509.RSA {
+		return true
+	}
+
+	return false
+}
+
+func (l *allowedRSAKeyModulus) Execute(c *x509.Certificate) *LintResult {
+	pubKey, ok := c.PublicKey.(*rsa.PublicKey)
+	if !ok {
+		return &LintResult{Status: Fatal, Details: "certificate public key was not an RSA public key"}
```

Comment:
Can you be more specific error here? Details should really only be included if they include more information than the name of the lint. Keys that fail this test are indeed valid RSA public keys.

Reply:
Please check reply above, can you propose something?

Reply:
I didn't realize this was a Fatal, I agree this looks fine. Apologies for missing that.